### DISCUSSION TITLE
Ability to parse and access form data in server middleware functions.

### DISCUSSION THREAD

**Discussion by christophior** [remix-run/remix#3575] [Q&A]
Hey all, reaching out to see how to possibly access the form data in an Express middleware that gets hit before we reach a remix action function. Currently when we attempt to parse the form data the action function in remix will not have any data since it has already been parsed. 

One possible workaround might be to parse the data and then have the data injected into the `context` object via [`getLoadContext()` ](https://remix.run/docs/en/v1/api/conventions#loader-context). Another possible solution could be to have a way to let remix know that the form data has already been parsed.

**Comment by sergiodxa** [comment]
You need to either pass the already parsed data as you said or clone the request in Express so you can read the body without making it impossible to read to Remix.

The whole issue is because request bodies could only be read once, you need to clone the request to read it more than once.

**Reply by christophior** [reply]
@sergiodxa any recommended way to clone the express request?

**Reply by sergiodxa** [reply]
I never did it in Express, I know in Remix you could do `requeset.clone()`, maybe check how the Express adapter works.
